broker-core/src/main/java/org/apache/qpid/server/model/NamedAddressSpace.java

```diff
@@ -48,8 +50,17 @@
 
     boolean registerConnection(AMQPConnection<?> connection,
                                final ConnectionEstablishmentPolicy connectionEstablishmentPolicy);
+
     void deregisterConnection(AMQPConnection<?> connection);
 
+    default ResourceAvailability requestConnectionSlot(AMQPConnection<?> connection, String userId)
```

Comment:
This leads to the sharing of the connection counters among VistualHosts, because broker connection limit providers are common for all VistualHosts. But, it is stated in documentation that VistualHosts are independent, hence every VistualHosts needs it's own ConnectionLimiter with independent connection counters. ConnectionLimitProvider works as factory that build a new independent limiter for each VirtualHost.